sdk/sdkservices/runtimes.go

```diff
@@ -88,12 +98,73 @@ func (rc *RunCallbacks) SafeCall(ctx context.Context, rid sdktypes.RunID, v sdkt
 	return rc.Call(ctx, rid, v, args, kwargs)
 }
 
-func (rc *RunCallbacks) SafePrint(ctx context.Context, rid sdktypes.RunID, text string) {
+func (rc *RunCallbacks) SafePrint(ctx context.Context, rid sdktypes.RunID, text string) error {
 	if rc == nil || rc.Print == nil {
-		return
+		return nil
+	}
+
+	return rc.Print(ctx, rid, text)
+}
+
+func (rc *RunCallbacks) SafeNewRunID() (sdktypes.RunID, error) { return sdktypes.NewRunID(), nil }
+
+func (rc *RunCallbacks) SafeSleep(ctx context.Context, rid sdktypes.RunID, d time.Duration) error {
```

Comment:
What's the purpose of all these `Safe*` functions?

Reply:
if callbacks are nil or the specific callback is nil, does not crash. I think i'll remove them.